keanu-project/src/main/java/io/improbable/keanu/tensor/validate/TensorValidator.java

```diff
@@ -0,0 +1,48 @@
+package io.improbable.keanu.tensor.validate;
+
+import java.util.function.Function;
+
+import io.improbable.keanu.tensor.Tensor;
+import io.improbable.keanu.tensor.bool.BooleanTensor;
+import io.improbable.keanu.tensor.validate.check.CustomTensorValueChecker;
+import io.improbable.keanu.tensor.validate.check.TensorValueChecker;
+import io.improbable.keanu.tensor.validate.check.TensorValueNotEqualsCheck;
+import io.improbable.keanu.tensor.validate.policy.TensorValidationPolicy;
+
+public class TensorValidator<U, T extends Tensor<U>> implements TensorValueChecker<T> {
+
+    public static <U, T extends Tensor<U>> TensorValidator<U, T> thatChecksFor(U v) {
```

Comment:
I found this really confusing! When I first read the tests I assumed that `thatChecksFor` was checking that they existed when in face it checks its not equal. Can we rename it to `checkValueIsNotPresent` or `checksIsNotValue` or something like that?

Reply:
changed to `thatExpectsNotToFind`

Reply:
nice